### PR TITLE
docker: allow IPFS_PROFILE to choose the profile for `ipfs init`

### PR DESCRIPTION
**Pull request by akaihola**
Added support for an `IPFS_PROFILE` environment variable in the `bin/container_daemon`. The variable, if present, is used with `ipfs init` to add a `--profile=$IPFS_PROFILE` argument which will apply to chosen profile to the newly created configuration.

This is useful for automatic IPFS deployments in hosted container environments, e.g. Kubernetes. In such situations, it's essential to use `--profile=server`.

Pull request helm/charts#7746 requires this change in order to add the `profile` configurable parameter to the IPFS Helm chart.

### CODE REVIEW

**Comment by akaihola** [comment]
It seems GitCop rejected my commit message as invalid, but I can't find the guidelines for commit messages. My message is 64 characters long, so that shouldn't be the issue. Advice, anyone?

**Comment by rob-deutsch** [comment]
@akaihola guidance is here: https://github.com/ipfs/go-ipfs/blob/master/contribute.md

The commit message is missing the trailers.

**Comment by akaihola** [comment]
@rob-deutsch, thanks for your explanation and sorry for missing those instructions.

I know corrected the commit message and force pushed the branch.

File: bin/container_daemon [hunk 1]
```diff
@@ -17,7 +17,11 @@ ipfs version
 if [ -e "$repo/config" ]; then
   echo "Found IPFS fs-repo at $repo"
 else
-  ipfs init
+  case $IPFS_PROFILE in
+    "") INIT_ARGS= ;;
+    *) INIT_ARGS=--profile=$IPFS_PROFILE ;;
+  esac
+  ipfs init $INIT_ARGS
```
[the remark this answers is not in the document]
Comment (akaihola) [reply bin/container_daemon]: Quoting the argument in `ipfs init "$INIT_ARGS"` actually breaks the script. But I did add quoting to the `case` statement.

**Comment by funkyfuture** [comment]
@Stebalien afaict, your remaining concern has been resolved (it is marked as outdated in the view), and you might approve this now.

**Comment by Stebalien** [comment]
@whyrusleeping @lgierth any objections?

**Comment by whyrusleeping** [comment]
Just for the docker stuff? Seems fine to me

**Comment by victorb** [comment]
Any chance of merging this? Makes infra a lot simpler (no need to have a separate Dockerfile to just override the profile), seems straightforward and been approved.

**Pull request merged**